base/bucket_gocb.go

```diff
@@ -952,8 +896,85 @@ func (bucket CouchbaseBucketGoCB) Update(k string, exp int, callback sgbucket.Up
 }
 
 func (bucket CouchbaseBucketGoCB) WriteUpdate(k string, exp int, callback sgbucket.WriteUpdateFunc) error {
-	LogPanic("Unimplemented method: WriteUpdate()")
-	return nil
+
+	bucket.singleOps <- struct{}{}
+	defer func() {
+		<-bucket.singleOps
+	}()
+
+	// The number of Couchbase Nodes that durable writes should be replicated to or persisted to
+	numNodesReplicateTo := uint(0)
+	numNodesPersistTo := uint(1)
+
+	maxCasRetries := 100000 // prevent infinite loop
+	for i := 0; i < maxCasRetries; i++ {
+
+		var value []byte
+		var err error
+		var writeOpts sgbucket.WriteOptions
+		var cas uint64
+
+		// Load the existing value.
+		gocbExpvars.Add("Update_Get", 1)
+		cas, err = bucket.Get(k, &value)
+		if err != nil && err != gocb.ErrKeyNotFound {
+			// Unexpected error, abort
+			return err
+		}
+
+		// Invoke callback to get updated value
+		value, writeOpts, err = callback(value)
+		if err != nil {
+			return err
+		}
+
+		if cas == 0 {
```

Comment:
I just noticed an issue while looking into this.

Currently we're calling:

```
if err != nil && err != gocb.ErrKeyNotFound {
```

but in my testing I found that this comparison doesn't work, since `gocb.ErrKeyNotFound` is actually a pointer and it doesn't seem to provide any guarantees that it will be returning a pointer to the same object.  I didn't actually look at the code, but from my testing that comparison approach wasn't reliable, which is why I switched to the string comparison approach.

I will push up a fix that uses the string comparison.